### ISSUE TITLE
Detect rustc sources in toolchain sysroot

### ISSUE THREAD

**Issue by camsteffen** [intellij-rust/intellij-rust#6862]
It should be fairly straightforward to port the changes from rust-analyzer/rust-analyzer#7643. This would be great for Rust contributors!

**Comment by Undin** [comment]
@camsteffen it's already implemented a long time ago.
What issue do you have?

**Comment by Undin** [comment]
@camsteffen oh, do you mean sources of `rustc` itself?

**Comment by camsteffen** [comment]
Yes. `rustc_*` modules do not resolve for the rust-clippy project.

**Comment by Undin** [comment]
Hm, interesting. How do such modules usually get to the sysroot?

**Comment by camsteffen** [comment]
Through the `rustc-dev` toolchain component.

**Comment by Undin** [comment]
@camsteffen Got it. Thanks for the clarification!

**Comment by orium** [comment]
@Undin If you can give me some high-level instructions on how to implement this I'm happy to give it a shot.
